src/jsrm/systems/planar_pcs_num.py

```diff
@@ -0,0 +1,1305 @@
+                integral_Jo = jnp.sum(Ws[:, None, None] * integrand_JoT_Jo, axis=0)
+                
+                integral = rho[i] * A[i] * integral_Jp + rho[i] * Ib[i] * integral_Jo
+                
+            elif integration_type == "gauss-kronrad":
+                rule = GaussKronrodRule(order=param_integration)
+                def integrand(s):
+                    J = jacobian_fn_xi(params, xi, s)
+                    Jp = J[:2, :]
+                    Jo = J[2:, :]
+                    return rho[i] * A[i] * Jp.T @ Jp + rho[i] * Ib[i] * Jo.T @ Jo
+
+                integral, _, _, _ = rule.integrate(integrand, l_cum[i], l_cum[i+1], args=())
+
+            elif integration_type == "trapezoid":
+                xs = jnp.linspace(l_cum[i], l_cum[i + 1], param_integration)
+                
+                J_all = vmap(lambda s: jacobian_fn_xi(params, xi, s))(xs)
+                Jp_all = J_all[:, :2, :]
+                Jo_all = J_all[:, 2:, :]
+
+                integrand_JpT_Jp = jnp.einsum("nij,nik->njk", Jp_all, Jp_all)
+                integrand_JoT_Jo = jnp.einsum("nij,nik->njk", Jo_all, Jo_all)
+                
+                integral_Jp = jscipy.integrate.trapezoid(integrand_JpT_Jp, x=xs, axis=0)
+                integral_Jo = jscipy.integrate.trapezoid(integrand_JoT_Jo, x=xs, axis=0) 
+                
+                integral = rho[i] * A[i] * integral_Jp + rho[i] * Ib[i] * integral_Jo
+                
+            return integral
+        
+        # Compute the cumulative integral
+        indices = jnp.arange(num_segments) 
+        integrals = vmap(compute_integral)(indices)
+        
+        B = jnp.sum(integrals, axis=0)
+        
+        return B
+        
+    @jit
+    def B_C_fn_xi(
+        params: Dict[str, Array], 
+        xi: Array, 
+        xi_d: Array
+    ) -> Tuple[Array, Array]:
+        """
+        Compute the mass / inertia matrix of the robot and the Coriolis / centrifugal matrix of the robot.
+
+        Args:
+            params (Dict[str, Array]): dictionary of robot parameters.
+            xi (Array): strain vector of the robot.
+            xi_d (Array): velocity vector of the robot.
+
+        Returns:
+            Tuple[Array, Array]: 
+                - B (Array): mass / inertia matrix of the robot.
+                - C (Array): Coriolis / centrifugal matrix of the robot.
+        """
+
+        # Compute the mass / inertia matrix
+        B = B_fn_xi(params, xi)
+
+        # Compute the Christoffel symbols
+        def christoffel_symbol(i, j, k):
+            return 0.5 * (
+                grad(lambda x: B[i, j])(xi)[k]
+                + grad(lambda x: B[i, k])(xi)[j]
+                - grad(lambda x: B[j, k])(xi)[i]
+            )
+
+        # Compute the Coriolis / centrifugal matrix
+        C = jnp.zeros_like(B)
+        for i in range(B.shape[0]):
+            for j in range(B.shape[1]):
+                for k in range(B.shape[0]):
```

Comment:
These nested Python-native for loops are really bad for compilation time - in particular when the dimension of the mass matrix becomes large (i.e., many segments).